core/shared/src/main/scala/org/http4s/Message.scala

```diff
@@ -232,6 +284,25 @@ sealed trait Message[+F[_]] extends Media[F] { self =>
 
   def mapK[F2[x] >: F[x], G[_]](f: F2 ~> G): SelfF[G] =
     self.change(entity = entity.translate(f))
+
+  /** Compiles the body stream to a single chunk and sets it as the
+    * body.  Replaces any `Transfer-Encoding: chunked` with a
+    * `Content-Length` header.  It is the caller's responsibility to
+    * assure there is enough memory to materialize the body.
+    */
+  def toStrict[F1[x] >: F[x]](implicit F: Concurrent[F1]): F1[SelfF[F1]] =
+    entity match {
+      case Entity.Empty =>
+        F.pure(self.covary[F1])
+      case Entity.Strict(chunk) =>
+        F.pure(self.withContentLength(`Content-Length`.unsafeFromLong(chunk.size.toLong)))
```

Comment:
One thing I'm not sure about - should we replace the `Content-Length` header here. I'm leaning that it's better to do.